motoman_driver/MotoPlus/MotionServer.c

```diff
@@ -286,6 +280,9 @@ int Ros_MotionServer_GetExpectedByteSizeForMessageType(SimpleMsg* receiveMsg, in
 	case ROS_MSG_MOTO_GET_DH_PARAMETERS:
 		expectedSize = minSize; //no additional data on the request
 		break;
+	case ROS_MSG_MOTO_SELECT_TOOL:
+		expectedSize = minSize + sizeof(SmBodySelectTool);
+		break;

```

Comment:
This seems to introduce a new message and some infrastructure, but it's unclear to me what the use-case is.

Is this needed for the PFL functionality (information about masses and inertias based on tool configuration)?

The rest of MotoROS is joint control only, so introducing a tool configuration setter seems unrelated.

Reply:
> Is this needed for the PFL functionality (information about masses and inertias based on tool configuration)?

Yes